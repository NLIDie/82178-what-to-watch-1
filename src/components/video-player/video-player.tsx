import * as React from "react";

// Types
import { TProps } from './video-player.types';

export function VideoPlayer(props: TProps): JSX.Element {
  const {
    muted = true,
    poster,
    src,
    onMouseOut,
    onMouseOver,
  } = props;

  return (
    <video
      width="100%"
      height="100%"
      src={src}
      poster={poster}
      muted={muted}
      onMouseOver={onMouseOver}
      onMouseOut={onMouseOut}
    />
  );
}
