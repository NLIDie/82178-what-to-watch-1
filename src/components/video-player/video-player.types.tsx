export type TProps = React.HTMLProps<HTMLVideoElement>;
