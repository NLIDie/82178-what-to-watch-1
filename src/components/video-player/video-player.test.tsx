import * as React from 'react';
import { create } from 'react-test-renderer';

// Components
import { VideoPlayer } from './video-player';

describe('Components/VideoPlayer', () => {
  it('Correct renderer the component', () => {
    const tree = create(
      <VideoPlayer
        muted
        poster="img/poster"
        src="video/src"
        onMouseOut={jest.fn()}
        onMouseOver={jest.fn()}
      />
    );

    expect(tree).toMatchSnapshot();
  })
})
