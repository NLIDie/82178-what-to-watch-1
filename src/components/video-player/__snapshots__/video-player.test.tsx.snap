// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/VideoPlayer Correct renderer the component 1`] = `
<video
  height="100%"
  muted={true}
  onMouseOut={[MockFunction]}
  onMouseOver={[MockFunction]}
  poster="img/poster"
  src="video/src"
  width="100%"
/>
`;
